scss/components/_button.scss

```diff
@@ -158,13 +158,23 @@ $button-transition: background-color 0.25s ease-out, color 0.25s ease-out !defau
 }
 
 /// Adds disabled styles to a button by fading the element, reseting the cursor, and disabling pointer events.
-@mixin button-disabled($color: $primary-color) {
+@mixin button-disabled(
+  $background: $primary-color,
+  $color: $button-color
+) {
   opacity: $button-opacity-disabled;
   cursor: not-allowed;
+  background-color: $background;
+  
+  @if $color == auto {
```

Comment:
Please put logic and variable definition in the top of the mixin, before declaring any property.